hs_core/templatetags/hydroshare_tags.py

```diff
@@ -49,6 +50,20 @@ def user_resource_labels(resource, user):
     return []
 
 
+@register.filter
+def get_user_privilege(resource, user):
+    user_privilege = resource.raccess.get_effective_user_privilege(user)
```

Comment:
This will generate one query per resource.

Reply:
I see, because of this query?
https://github.com/hydroshare/hydroshare/blob/develop/hs_access_control/models/resource.py#L338
`p = UserResourcePrivilege.objects.get(resource=self.resource, user=this_user)`

Hmm bit of a Catch22: not using annotations causes 1 additional query. But using annotations has issues when union the annotated query sets.

Reply:
Since we are not loading the shared resources by default, I think it is OK to have this additional query. Once we optimize other queries generated from the template, it would be a substantial improvement for page load.